utils/cache.js

```diff
@@ -59,25 +69,87 @@ const cachePool = (opt = { maximumSize: CACHE_SIZE_MB }) => {
     }
   };
 
-  return { get, set, hits, cacheStore };
+  return { get, set, evict, hits, cacheStore };
+};
+/**
+ * A MultiMap implementation where each key maps to set of unique values
+ *
+ */
+const cachedKeysStore = () => {
+  const keyStore = new Map();
+
+  /**
+   * Returns set of values mapped to given key
+   * @param {string} modelKey key for the map
+   * @returns {Set} set of values
+   */
+  const getCachedKeys = (modelKey) => {
+    if (keyStore.has(modelKey)) {
+      return keyStore.get(modelKey);
+    }
+    return new Set();
```

Comment:
- as we are returning an array inside the if condition should we also not return an empty array as the base case?

 - when doing a default return it's generally followed that we handle the false case first. 

Example:-
```js
if (!keys){
    ...do something 
}

<We can now handle our happy case>
```

What do you think?

Reply:
I agree with this for most cases, I think it's a good way to avoid unnecessary nesting in code. 
But in this context I don't feel that's the right approach.  As we are only interested in one case here and that's if the given key is present in the data store. 

for eg. this is how it would look if we handle the negative case first
```javascript
  const getCachedKeys = (modelKey) => {
     if (!keyStore.has(modelKey)) {
            return new Set();
     }
     // handle unnecessary edge cases where we should also return an empty set.
     
     if (keyStore.has(modelKey)) {      // still we cannot remove this check
       return keyStore.get(modelKey);
     }
   }
     
```

Reply:
Once you do this check and ensure the key is present

```js
if (!keyStore.has(modelKey)) {
        return new Set();
}
```

Why do we have to do this check again ? 

```js
if (keyStore.has(modelKey)) {
        ....
}
```

Reply:
- It is a coding style that I generally see in many places  where we handle the case of data not being present / incorrect / invalid this then makes sure that any operation being performed after the check has the correct data we want to use. 

A close example to this you would see in typescript repo's where we handle the `undefined`/`null` case at the top level of the function and have our happy path after that. 

This also make us think of the error cases before we start performing the actual operation.

Reply:
added